src/librustc_codegen_ssa/mir/rvalue.rs

```diff
@@ -380,6 +380,25 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
                 })
             }
 
+            mir::Rvalue::AddressOf(mutability, ref place) => {
```

Comment:
Since this branch is near-identical with the one above, imo it should be de-duped until such time that they diverge.